Document cuPointerGetAttribute workaround for Nvidia driver 590 (#410)
eth-cscs/cscs-docs: docs/software/communication/cray-mpich.md

### Existing Issues

[](){#ref-communication-cray-mpich-cupointergetattribute-slowdown}
#### Slow host buffer communication with Nvidia driver version 590 and later

Starting with NVIDIA driver version 580, calls to `cuPointerGetAttribute` with host-allocated buffers became significantly slower.
Cray MPICH relies on `cuPointerGetAttribute` to determine whether to use host or device code paths for communication.
For applications that communicate many small host buffers this can introduce significant slowdowns.

For applications that only communicate host buffers, explicitly disabling GPU support for Cray MPICH will allow it to completely skip the host/device buffer check:
```bash
export MPICH_GPU_SUPPORT_ENABLED=0
```

For applications that use both host and device buffers for communication, the following shared library can be preloaded to almost fully restore performance:

```bash
export LD_PRELOAD=/capstor/store/cscs/cscs/public/temp/cuptrgetattr_override.so
```

> The library overrides the `cuPointerGetAttribute` function with an implementation based on `cuPointerGetAttributes` which is faster than the implementation provided in driver version 580.

Reviewer: ```suggestion
... in driver version >=580.
```

Author: ```suggestion
The library overrides the `cuPointerGetAttribute` function with an implementation based on `cuPointerGetAttributes` which is faster than the implementation provided in driver version 580 and later.
```

!!! warning
    The shared library is provided without any guarantees about compatibility with the `cuPointerGetAttribute` implementation provided by NVIDIA.
    The library has been tested to work in Cray MPICH use cases, but may not be provide exactly the same behaviour as the real implementation in terms of error reporting and checks.

!!! warning

>     The shared library may be removed without advance notice.

Reviewer: ```suggestion
    When the issue is resolved upstream the shared library will be removed without advance notice.
```
